### PR TITLE
Automation: Verify release artifacts on grafana.com

### PR DESCRIPTION
**Pull request by baldm0mma**
**What is this feature?**

This PR verifies all release artifacts that are available for download on grafana.com.

This creates a step in the pipeline called `verify_grafanacom_step` that installs the GCOM util, then runs a shell script that gathers the download links for all grafana artifacts on grafana.com, then hits those endpoints to see if they return a 200 resp - i.e., are available for download.

**Why do we need this feature?**

CI automation makes everyone smile

**Who is this feature for?**

All humans

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-release/issues/1040

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: scripts/drone/steps/lib.star [hunk 1]
```diff
@@ -1255,3 +1255,30 @@ def slack_step(channel, template, secret):
...
+            "curl -L -o /usr/local/bin/gcom https://github.com/grafana/gcom/releases/latest/download/gcom-linux-amd64",
+            "chmod +x /usr/local/bin/gcom",
+
+            # Authenticate gcloud
+            "printenv GCP_KEY > /tmp/key.json",
+            "gcloud auth activate-service-account --key-file=/tmp/key.json",
+            "./drone/verify-grafanacom.sh",
+
+            # Remove the temporary key file
+            "rm /tmp/key.json",
```
Comment (kminehart) [inline scripts/drone/steps/lib.star:1281]:
```suggestion

            # Remove the temporary key file
            "rm /tmp/key.json",
```

No sense in wasting CPU cycles deleting files that will be deleted later anyways :P

**Review by kminehart — APPROVED** [review]
I think it'll work. There may be some time between when `publish_grafanacom` happens and when it's available on the website... maybe we should account for that?

File: scripts/drone/steps/lib.star (hunk 1 above)
Comment (baldm0mma) [reply scripts/drone/steps/lib.star]: Touché. Removed.

**Comment by baldm0mma** [comment]
> I think it'll work. There may be some time between when `publish_grafanacom` happens and when it's available on the website... maybe we should account for that?

Hmm, good point. I could introduce a retry mechanism in the `verify_grafanacom_step` that would allow the verification step to retry the verification process a few times if the artifacts are not immediately available? Just a simple for loop that runs the `verify-grafanacom.sh` script up to 5? times, and retries every 60? seconds on failure?

File: scripts/drone/events/release.star [hunk 2]
```diff
@@ -211,6 +213,19 @@ def publish_packages_pipeline():
...
+            trigger={
+                "event": ["promote"],
+                "target": "verify-grafanacom-artifacts"
+            },
+            steps=[
+                verify_grafanacom_step(),
+            ],
+            depends_on=[
+                "publish-grafanacom",
+            ],
```
Comment (kminehart) [inline scripts/drone/events/release.star:227]:
```suggestion
```

**Review by kminehart — CHANGES_REQUESTED** [review]

File: scripts/drone/steps/lib.star [hunk 3]
```diff
@@ -1152,6 +1152,36 @@ def publish_grafanacom_step(ver_mode):
...
+            "gcloud auth activate-service-account --key-file=/tmp/key.json",
+
+            # Make the following script executable.
+            "chmod +x ./drone/verify-grafanacom.sh",
+
+            # There may be a slight lag between when artifacts are uploaded to Google Storage,
+            # and when they are available on the website. This sould account for that discrepancy.
+            "for i in {{1..5}}; do ./drone/verify-grafanacom.sh && break || sleep 60; done",
+        ],
+        "depends_on": ["publish-grafanacom"],
```
Comment (kminehart) [inline scripts/drone/steps/lib.star:1182]:
```suggestion
        "depends_on": depends_on,
```

File: scripts/drone/steps/lib.star [hunk 4]
```diff
@@ -1152,6 +1152,36 @@ def publish_grafanacom_step(ver_mode):
         ],
     }
 
+def verify_grafanacom_step():
```
Comment (kminehart) [inline scripts/drone/steps/lib.star:1155]:
```suggestion
def verify_grafanacom_step(depends_on=["publish-grafanacom"]):
```

File: scripts/drone/events/release.star [hunk 5]
```diff
@@ -211,6 +213,19 @@ def publish_packages_pipeline():
...
 
     return [
+        pipeline(
+            name="verify-grafanacom-artifacts",
+            trigger={
+                "event": ["promote"],
+                "target": "verify-grafanacom-artifacts"
+            },
+            steps=[
+                verify_grafanacom_step(),
```
Comment (kminehart) [inline scripts/drone/events/release.star:223]:
```suggestion
                verify_grafanacom_step(depends_on=["init-enterprise"]),
```

**Review by kminehart — CHANGES_REQUESTED** [review]

File: scripts/drone/events/release.star [hunk 6]
```diff
@@ -211,6 +213,16 @@ def publish_packages_pipeline():
...
```
Comment (kminehart) [inline scripts/drone/events/release.star:223]:
```suggestion
                verify_grafanacom_step(depends_on=[]),
```

**Review by kminehart — CHANGES_REQUESTED** [review]

File: scripts/drone/steps/lib.star [hunk 7]
```diff
@@ -1152,6 +1152,36 @@ def publish_grafanacom_step(ver_mode):
...
+            "apk add curl",
+
+            # Download/install gcom utility, and make it executable.
+            "curl -L -o /usr/local/bin/gcom https://github.com/grafana/gcom/releases/latest/download/gcom-linux-amd64",
+            "chmod +x /usr/local/bin/gcom",
+
+            # Authenticate gcloud.
+            "printenv GCP_KEY > /tmp/key.json",
+            "gcloud auth activate-service-account --key-file=/tmp/key.json",
+
```
Comment (kminehart) [inline scripts/drone/steps/lib.star:1174]:
```suggestion
```

File: scripts/drone/verify-grafanacom.sh [hunk 8]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+version=${1:-$TAG}
+
+base_url="/downloads/grafana/versions/$version"
```
Comment (kminehart) [inline scripts/drone/verify-grafanacom.sh:5]:
```suggestion
base_url="https://grafana.com/api/downloads/grafana/versions/$version"
```

File: scripts/drone/verify-grafanacom.sh [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+# Construct the URL based on the provided version and edition
+if [ "$EDITION" = "enterprise" ]; then
+    url="/downloads/grafana-enterprise/versions/$version"
```
Comment (kminehart) [inline scripts/drone/verify-grafanacom.sh:9]:
```suggestion
    url="https://grafana.com/api/downloads/grafana-enterprise/versions/$version"
```

File: scripts/drone/verify-grafanacom.sh [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+else
+    url="/downloads/grafana/versions/$version"
```
Comment (kminehart) [inline scripts/drone/verify-grafanacom.sh:11]:
```suggestion
    url="https://grafana.com/api/downloads/grafana/versions/$version"
```

File: scripts/drone/verify-grafanacom.sh [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+fi
+
+# Make a request to the GCOM API to retrieve the artifacts for the specified version. Exit if the request fails.
+if ! artifacts=$(gcom "$url"); then
```
Comment (kminehart) [inline scripts/drone/verify-grafanacom.sh:15]:
```suggestion
if ! artifacts=$(curl "$url"); then
```

**Review by kminehart — CHANGES_REQUESTED** [review]

**Review by kminehart — CHANGES_REQUESTED** [review]

File: scripts/drone/steps/lib.star [hunk 12]
```diff
@@ -1152,6 +1152,23 @@ def publish_grafanacom_step(ver_mode):
...
 
+def verify_grafanacom_step(depends_on=["publish-grafanacom"]):
+    return {
+        "name": "verify-grafanacom",
+        "image": images["node"],
+        "commands": [
+            # Download and install `curl` - it isn't available inside of the `node:{version}-alpine` docker image.
+            "apk add curl",
+            # Make the following script executable.
+            "chmod +x ./scripts/drone/verify-grafanacom.sh",
```
Comment (kminehart) [inline scripts/drone/steps/lib.star:1163]:
```suggestion
            "apk add curl bash",
```

**Review by kminehart — CHANGES_REQUESTED** [review]

**Review by kminehart — APPROVED** [review]

**Pull request merged**